include/uprotocol-cpp/uri/builder/BuildUResource.h

```diff
@@ -0,0 +1,128 @@
+#include <string>
+#include <array>
+#include <string_view>
+#include <arpa/inet.h>
+#include <spdlog/spdlog.h>
+#include <uprotocol-cpp/uri/tools/Utils.h>
+#include "uprotocol/uri.pb.h"
+
+namespace uprotocol::uri {
+    class BuildUResource {
+        uprotocol::v1::UResource resource_;
+
+    public:
+        BuildUResource() { resource_.Clear();}
+        
+        auto setName(const std::string &name) -> BuildUResource & {
+            if (isBlank(name)) {
+                spdlog::error("UResource name cannot be empty");
+            } else {
+                resource_.set_name(name);
+            }
+            return *this;
+        }
+        
+        auto setInstance(const std::string &instance) -> BuildUResource & {
+            if (isBlank(instance)) {
+                spdlog::error("UResource instance cannot be empty");
+            } else {
+                resource_.set_instance(instance);
+            }
+            return *this;
+        }
+        auto setMessage(const std::string &message) -> BuildUResource & {
+            if (isBlank(message)) {
+                spdlog::error("UResource message cannot be empty");
+            } else {
+                resource_.set_message(message);
+            }
+            return *this;
+        }
+        auto setID(const uint32_t &id) -> BuildUResource & {
+            if (0 == id) {
+                spdlog::error("UResource id cannot be 0");
+            } else {
+                resource_.set_id(id);
+            }
+            return *this;
+        }
+    
+        auto setRpcRequest(const std::string &method) -> BuildUResource & {
+            if (isBlank(method)) {
+                spdlog::error("UResource method cannot be empty");
+            } else {
+                resource_.set_name("rpc");
+                resource_.set_instance(method);
+            }
+            return *this;
+        }
+        auto setRpcRequest(const std::string &method, uint32_t id) -> BuildUResource & {
+            if (isBlank(method) || 0 == id) {
+                spdlog::error("UResource method cannot be empty");
+            } else {
+                setName("rpc");
+                setInstance(method);
+                setID(id);
+            }
+            return *this;
+        }
+        auto setRpcRequest(const uint32_t id) -> BuildUResource & {
+            if (0 == id) {
+                spdlog::error("UResource id cannot be 0");
+            } else {
+                setName("rpc");
+                setID(id);
+            }
+            return *this;
+        }
+        auto setRpcResponse() -> BuildUResource & {
+            setName("rpc");
+            setInstance("response");
+            setID(0);
+            return *this;
+        }
+        
+        auto build() const -> uprotocol::v1::UResource {
+            return resource_;
+        }
+    };
+    
+    [[nodiscard]] [[maybe_unused]] auto static isRPCMethod(const uprotocol::v1::UResource &resource) -> bool {
```

Comment:
Also the ID needs to be <1000 per the spec for RPCs and topics are > 1000.  Can you add a isRpcResponse() as well?